src/pages/SupplierRequisitionPage.js

```diff
@@ -320,38 +343,69 @@ export class SupplierRequisitionPage extends React.Component {
     }
   };
 
+  renderThresholdMOSToggle = () => {
+    const { useThresholdMOS } = this.state;
+    const onPress = () => this.setState({ useThresholdMOS: !useThresholdMOS }, this.refreshData);
+    const toggleProps = [
+      {
+        text: 'Hide over stocked',
+        isOn: useThresholdMOS,
+        onPress,
+      },
+      {
+        text: 'Show over stocked',
+        isOn: !useThresholdMOS,
+        onPress,
+      },
+    ];
+    return <ToggleBar style={globalStyles.toggleBar} toggles={toggleProps} />;
+  };
+
   renderButtons = () => {
     const { requisition } = this.props;
+    const { program } = requisition;
 
     return (
       <View style={globalStyles.pageTopRightSectionContainer}>
         <View style={globalStyles.verticalContainer}>
           <PageButton
-            style={globalStyles.topButton}
-            text={buttonStrings.create_automatic_order}
-            onPress={this.onCreateAutomaticOrder}
-            isDisabled={requisition.isFinalised}
-          />
-          <PageButton
-            style={globalStyles.leftButton}
+            style={{
+              ...globalStyles.topButton,
+              ...(program ? { marginLeft: 5 } : {}),
+            }}
             text={buttonStrings.use_suggested_quantities}
             onPress={this.onUseSuggestedQuantities}
             isDisabled={requisition.isFinalised}
           />
+
+          {program && this.renderThresholdMOSToggle()}
```

Comment:
Here again i think, program && requisition.thresholdMOS // if requisition.thresholdMOS is missing or is 0, don't show the toggle

Reply:
fixed 47e28093d7fd4a9e6496fccbdab1bfb94d9a2220